lib/matplotlib/legend.py

```diff
@@ -596,18 +600,24 @@ def __init__(
                     try:
                         color = getattr(handle, getter_name)()
                         if isinstance(color, np.ndarray):
-                            if (
-                                    color.shape[0] == 1
-                                    or np.isclose(color, color[0]).all()
+                            if color.size == 0:
+                                continue
+                            elif (
+                                color.shape[0] == 1
+                                or np.isclose(color, color[0]).all()
                             ):
                                 text.set_color(color[0])
                             else:
                                 pass
+                        elif cbook._str_lower_equal(color, 'none'):
+                            continue
+                        elif mpl.colors.to_rgba(color)[3] == 0:
+                                continue
                         else:
                             text.set_color(color)
                         break
                     except AttributeError:
-                        pass
+                        continue
```

Comment:
AFAICT, the `AttributeError` is only about the `getattr`, so this `try` block is starting to get very much too long. Can we adjust that a bit so that only that line is in the `try`? This also reduces the wrapping a bit.